sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/jsonschema/JsonSchemaResourceIntegrationTest.java

```diff
@@ -0,0 +1,267 @@
+    }
+
+    @Test
+    @Order(3)
+    public void keyspaceDoesNotExist() {
+      String body =
+          """
+                {\"$schema\": \"https://json-schema.org/draft/2019-09/schema\"}
+              """;
+
+      given()
+          .contentType(ContentType.JSON)
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .body(body)
+          .put(BASE_PATH, "notexist", DEFAULT_COLLECTION)
+          .then()
+          .statusCode(404)
+          .body("code", is(404))
+          .body("description", is("Unknown namespace notexist, you must create it first."));
+    }
+
+    @Test
+    @Order(4)
+    public void invalidJsonSchema() {
+      String body =
+          "{\n"
+              + "  \"$schema\": \"https://json-schema.org/draft/2019-09/schema\",\n"
+              + "  \"type\": \"object\",\n"
+              + "  \"properties\": {\n"
+              + "    \"something\": { \"type\": \"strin\" }\n"
+              + "  }\n"
+              + "}";
+
+      given()
+          .contentType(ContentType.JSON)
+          .body(body)
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .when()
+          .put(BASE_PATH, DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+          .then()
+          .statusCode(400)
+          .body("code", is(400))
+          .body(
+              "description",
+              is(
+                  "[error]: \"strin\" is not a valid primitive type (valid values are: [array, boolean, integer, null, number, object, string]); "));
+    }
+
+    @Test
+    @Order(5)
+    public void noPayload() {
+      given()
+          .contentType(ContentType.JSON)
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .when()
+          .put(BASE_PATH, DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+          .then()
+          .statusCode(400)
+          .body("code", is(400))
+          .body("description", is("Request invalid: json schema not provided."));
+    }
+
+    @Test
+    @Order(6)
+    public void notDocumentTable() {
+      given()
+          .contentType(ContentType.JSON)
+          .body("{}")
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .when()
+          .put(BASE_PATH, "system", "peers")
+          .then()
+          .statusCode(400)
+          .body("code", is(400))
+          .body(
+              "description",
+              is(
+                  "The database table system.peers is not a Documents collection. Accessing arbitrary tables via the Documents API is not permitted."));
+    }
+  }
+
+  @Nested
+  @Order(2)
+  class GetJsonSchema {
+
+    @Test
+    @Order(1)
+    public void happyPath() {
+      given()
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .when()
+          .get(BASE_PATH, DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+          .then()
+          .statusCode(200)
+          .body("schema", Matchers.nullValue());
```

Comment:
:warning: :warning: Wait this is wrong, why are you not getting the schema back, you wrote in the `PutJsonSchema.happyPath`

Reply:
oh I didnt even think about the fact that state isn't cleaned between tests...crap there is something pretty wrong here